mitodl/salt-ops
# update ocw next build script to use ocw-hugo-themes

#### What are the relevant tickets?
Closes https://github.com/mitodl/salt-ops/issues/1409

#### What's this PR do?
This PR updates the webhook publish script to build the OCW Next site using the new `ocw-hugo-themes` repository as a sole source for webpack artifacts and Hugo templates.

#### How should this be manually tested?
I'm not familiar with how things like this are usually tested here, but I imagine we want to try deploying to RC and see what happens.  This PR is currently blocked by https://github.com/mitodl/ocw-hugo-themes/pull/6

#### Any background context you want to provide?
There are a few things about this that are currently not super optimal that will be addressed in the future.  With the way the Hugo module system works, you're supposed to be able to just put something like this in your `config.toml` file and when you run `hugo mod get -u` the modules will be automatically fetched from github:

```
[module]
  [[module.imports]]
    path = "github.com/mitodl/ocw-hugo-themes/base-theme"
  [[module.imports]]
    path = "github.com/mitodl/ocw-hugo-themes/www"
  [[module.mounts]]
    source = "archetypes"
    target = "archetypes"
  [[module.mounts]]
    source = "assets"
    target = "assets"
  [[module.mounts]]
    source = "data"
    target = "data"
  [[module.mounts]]
    source = "layouts"
    target = "layouts"
  [[module.mounts]]
    source = "static"
    target = "static"
```

The issue with this and our current release strategy is that our releases happen from the `release` branch.  When you run `hugo mod get -u` for the first time with these entries in your `config.toml` file, a `go.mod` file like this will be created:

```
module github.com/mitodl/ocw-www

go 1.16

require (
	github.com/mitodl/ocw-hugo-themes/base-theme v0.0.0-20210428202616-8f3bb7aa2c95 // indirect
	github.com/mitodl/ocw-hugo-themes/www v0.0.0-20210428202616-8f3bb7aa2c95 // indirect
)
```

As you can see, Hugo automatically inserts a "pseudo version" string to the end made out of a version number, date and commit hash.  Each time you run `hugo mod get -u` it will grab the latest version tag.  Initially, I don't think we want to use it this way as until Doof is writing version tags into the repo on release, it will just pull the latest version from the `main` branch.  So, the way I have done this is to manually pull the `release` branch of `ocw-hugo-themes` and use the `replace` directive in the various `go.mod` files to manually point it at the themes repo cloned on the `release` branch.
